cmake/SpectreCheckTargetDependencies.cmake

```diff
@@ -0,0 +1,652 @@
+          )
+        # Sets TARGET_DEPS
+        _get_deps_for_spectre_includes(${TARGET_NAME} "${LIST_OF_ALL_TARGETS}"
+          "${SPECTRE_INCLUDES}" "${TARGET_HEADER_FILES}" ${RECURSE_THROUGH_INCLUDES})
+        # "Pop" last element by setting the property to what it was at the start of
+        # the function call.
+        set_property(
+          GLOBAL PROPERTY GET_DEPS_FOR_SPECTRE_INCLUDES_PATH
+          ${GET_DEPS_FOR_SPECTRE_INCLUDES_PATH}
+          )
+        # Get the dependencies of the header file, since these are all
+        # also PRIVATE dependencies of the target.
+        foreach(_TARGET ${TARGET_DEPS})
+          if(NOT ${_TARGET} IN_LIST _TARGET_DEPS)
+            list(APPEND _TARGET_DEPS ${_TARGET})
+          endif(NOT ${_TARGET} IN_LIST _TARGET_DEPS)
+        endforeach(_TARGET ${TARGET_DEPS})
+        continue()
+      else(RECURSE_THROUGH_INCLUDES)
+        continue()
+      endif(RECURSE_THROUGH_INCLUDES)
+    endif(${SPECTRE_INCLUDE} IN_LIST TARGET_HEADER_FILES)
+
+    unset(_TARGET_WITH_INCLUDE)
+    foreach(DEP_TARGET ${LIST_OF_ALL_TARGETS})
+      if(${DEP_TARGET} STREQUAL ${TARGET_NAME})
+        continue()
+      endif(${DEP_TARGET} STREQUAL ${TARGET_NAME})
+
+      # Get the header files of DEP_TARG and see if any of the
+      # includes are part of DEP_TARGET.
+      get_target_headers(${DEP_TARGET} _DEP_TARGET_HEADERS)
+      if(${SPECTRE_INCLUDE} IN_LIST _DEP_TARGET_HEADERS)
+        set(_TARGET_WITH_INCLUDE ${DEP_TARGET})
+        break()
+      endif(${SPECTRE_INCLUDE} IN_LIST _DEP_TARGET_HEADERS)
+    endforeach(DEP_TARGET ${LIST_OF_ALL_TARGETS})
+    if(NOT _TARGET_WITH_INCLUDE)
+      message(FATAL_ERROR
+        "No known targets supply the file ${SPECTRE_INCLUDE} included in "
+        "file ${HEADER_OR_SOURCE} of target ${TARGET_NAME}."
+        )
+    endif(NOT _TARGET_WITH_INCLUDE)
+
+    if(NOT ${_TARGET_WITH_INCLUDE} IN_LIST _TARGET_DEPS)
+      list(APPEND _TARGET_DEPS ${_TARGET_WITH_INCLUDE})
+    endif(NOT ${_TARGET_WITH_INCLUDE} IN_LIST _TARGET_DEPS)
+  endforeach(SPECTRE_INCLUDE ${SPECTRE_INCLUDES_OF_FILE})
+
+  # "Return" TARGET_DEPS by setting it in PARENT_SCOPE
+  set(TARGET_DEPS ${_TARGET_DEPS} PARENT_SCOPE)
+endfunction(_get_deps_for_spectre_includes TARGET_NAME LIST_OF_ALL_TARGETS
+  SPECTRE_INCLUDES_OF_FILE TARGET_HEADER_FILES RECURSE_THROUGH_INCLUDES)
+
+# Returns the dependencies for the target
+#
+# Arguments
+#
+# TARGET_NAME:
+#     the target for which to check the link libraries/dependencies
+# LIST_OF_ALL_TARGETS:
+#     a list of all the targets that might be dependencies
+#     of the target ${TARGET_NAME}
+# LIST_OF_FILES:
+#     a list of the files for which to get the dependencies. Can be
+#     header/tpp or source files.
+# RECURSE_INCLUDES:
+#     if TRUE then the dependencies of the target's included header files
+#     will also be added. This is used when source files are passed because
+#     the dependencies of the header files must be marked as PUBLIC
```

Comment:
It would be helpful, as you've done in many other cases, to note how this returns by setting something at parent scope.